### ISSUE TITLE
No notification for new messages in "Kontaktanfragen" (Contact requests?)

### ISSUE THREAD

**Issue by csb0730** [deltachat/deltachat-android#149]
If a new message arrives in "Kontaktanfragen" item then no user notification is set/done.

So You never get aware for new mails here except You check it manually!
Furthermore You don't see later when it arrived.

Bug or feature?

Proposal:
- do a notification for this item/special inbox too like it is done for other chats/groups
- rename "Contact requests" to "Other messages" or similar and move it from settings menu
   to main screen.

**Comment by Ampli-fier** [comment]
You can enable this in the advanced settings menu. You will see these messages in the chatlist then. I would also prefer this to be the default.

**Comment by csb0730** [comment]
Hi Ampli-fier,

setting changed and now I see "Contact requests" in chatlist.
This works fine  :)

But,
I can confirm now that even with this settings change there is still no notification set for
new messages in "Contact requests".

Could be a bug?

**Comment by Ampli-fier** [comment]
For me, this works fine. I do get notifications.
Do you have logcat info available for the developer team?

**Comment by csb0730** [comment]
Hi ampli-fier,

unfortunately i have no logcat. I don't know how to do this.

I made some investigations in this direction but this seams not possible for a non developper!?

csb0730
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by Ampli-fier** [comment]
There are many ways to get logact info.
Probably the easiest way is to use an [app on your device](https://github.com/ukanth/afwall/wiki/HOWTO-Report-Bug#how-do-i-capture-my-logcat).
I prefer to enable USB debugging and connect to my phone with a Terminal program like puttytray or kitty (for Windows).
Some random reading in [german ](https://www.android-user.de/erste-schritte-mit-usb-debugging-und-adb/) or in [english](https://logmatic.io/blog/a-how-to-guide-to-debugging-with-android-logcat/).

I used `logcat -v time` and got these lines - even though they contain an error, notifications get displayed correctly:
```
08-11 21:48:07.540 I/DeltaChat(23466): T4: IDLE has data.
08-11 21:48:07.982 I/DeltaChat(23466): T4: 1 mails read from "INBOX" with 0 errors.
08-11 21:48:08.089 I/DeltaChat(23466): T4: IDLE start...
08-11 21:48:08.200 E/ActivityThread(23466): Failed to find provider info for com.teslacoilsw.notifier
08-11 21:48:08.203 W/System.err(23466): java.lang.IllegalArgumentException: Unknown URL content://com.teslacoilsw.notifier/unread_count
08-11 21:48:08.217 W/System.err(23466):         at android.content.ContentResolver.insert(ContentResolver.java:1203)
08-11 21:48:08.217 W/System.err(23466):         at com.b44t.messenger.NotificationsController$5.run(NotificationsController.java:753)
08-11 21:48:08.217 W/System.err(23466):         at android.os.Handler.handleCallback(Handler.java:739)
08-11 21:48:08.217 W/System.err(23466):         at android.os.Handler.dispatchMessage(Handler.java:95)
08-11 21:48:08.217 W/System.err(23466):         at android.os.Looper.loop(Looper.java:135)
08-11 21:48:08.217 W/System.err(23466):         at com.b44t.messenger.DispatchQueue.run(DispatchQueue.java:98)
```

You do get notifications for Chats, right?

**Comment by csb0730** [comment]
Thanks for this hints. I'll investigate in that.

But meanwhile I see that there are sometimes notifications and sometimes not.
The notification process seems not to be stable.

I want to distinguish between two situations:
  a) after switching the phone longer time into aircraft mode and just activate it and
  b) normal operation with activated connection.

Additionally I have DC and K9 (pop3) at the same phone and use it in parallel (same mailaccount).
Maybe there is a relation because in situation a) k9 querys new mails immediately after activating phone from aircraft mode!

An interesting question is: How is a new message detected? By receiving really at phone or with another mechanism? Maybe a developer (Björn?) is able to bring more light in this questions before we make extensive research?

To answer last question: in general I get notifications for chats, yes :)

**Comment by csb0730** [comment]
Ok, some news:

I disabled mail sync at k9 mail (pop3) at the same phone and mail account DC uses. So now I only have DC getting new mails for this account and no other mail client. Even while this period I prevent using GMX webmail while testing period!
Result shows, that now notifications for new messages in DC seems to come correct! So there seems to be an interference of other mail clients and DC for the same account (over the mail server)!

By the way, K9 tags always new mails as new at local client, regardless if using other clients for the same account or not. This leads to the question: What's the difference in recognizing mechanism?

I suspect that DC uses a "read" flag from mailserver?

**Comment by csb0730** [comment]
Regarding logcat:
With Debian Linux it's possible to install "adb" by standard repositorys (Synaptic)
No search required. Maybe with other distro's too?
 :-)

**Comment by Ampli-fier** [comment]
Looks like you found the root cause by yourself.

DC uses another mail protocol to interact with your mailbox: [IMAP](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol). Your phone (DC app) gets notified immediately about new messages to your inbox by [Push-IMAP](https://en.wikipedia.org/wiki/Push-IMAP).
I'm not an expert for POP3 protocol, but it looks like with POP3 the client app only checks, if the message if locally available or not, message states are ignored.
Using POP3 and IMAP (by using DC) at the same time is not a good idea, read [here](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol#Multiple_clients_simultaneously_connected_to_the_same_mailbox).
Better use IMAP in K9.

**Comment by csb0730** [comment]
Ok, if this is really the root cause, then I see the danger that some people who are testing DC run into the same situation than me. They will simply see that notification is not working properly or they will see some messages much later and don't know why!
I assume that many people will use their existing mail account with DC and this means that there will be always a secondary existing mail application. Whether k9 at the phone or with another one at PC.

So this means at the end, that only with an individual account for DC message notification is reliable!
This is not so good :-/

Then IMHO the best would be to use (at least additionally) the first arrival of a message at DC to detect new message and doing the user notification!
This would lead to two situations:
1) message has NO "read" flag from server -> new message (do notification)
2) message has "read" flag from server, but new in local DC -> new message (do notification)

Case 2) leads in the moment to the situation -> new message but (NO notification)!

Maybe introduce a user setting for this?
Any comments welcome!

**Comment by Ampli-fier** [comment]
When I have read a mail in my INBOX with K9 (using IMAP) then I do not get the mail/message at all. DC seems to ignore it. I think this is ok as this is not a new contact request.
What you describe sounds different and I wonder if this is because you use POP3 protocol.
Behaviour for case 2 would be correct in my opinion, because when **you** have read the message then why would you need another notification?

**Comment by csb0730** [comment]
Maybe there is a misunderstanding:
Case 2) currently means that both mail clients are working in parallel for same account/mail, I DON'T read it and it's fetched by both of them! First fetch wins?
Maybe there is a dependency what mail provider (mail server) is handling the account.

What I described in "2. message has ..." is a proposal from my side to produce a clear behaviour of DC for this situation.

**Comment by csb0730** [comment]
Ok, for a while I'm testing now with K9 and DC at same mail account and both with IMAP access. POP3 I disabled for K9.
This configuration shows a consistent notification behavior. The first program which shows a message tags the message as read and the second one takes the notification for that message away.

So I suspect that the described problem only occurs if, I'm
- using K9 with POP3 in parallel with DC (IMAP)
- and after switching the phone from "aircraft mode" to normal operation.

In this case K9 syncs immediately after switching back from aircraft mode. This is more quick than DC gets first push message from server. So there is the situation where some messages are tagged as read from K9 POP3 request before arriving at DC.

And exactly for these messages no notification is set now in DC!

Now: What to do?
- User setting/option (my comment from 21 Aug) or
- let it as it is or
- other options?

**Comment by testbird** [comment]
The option would be "Notifications for new messages already read elsewhere".

Not only usefull with concurrent pop3 clients, but also with multiple persons working with a shared inbox.

**Comment by csb0730** [comment]
I think an option like
-    "Notifications for new messages already read elsewhere" or
-    "Notifications for new local messages"

should be introduced.

@r10s: You have seen that issue?

**Comment by r10s** [comment]
Yes, I have seen this, I think the original, starting issue is covered by this one: https://github.com/deltachat/deltachat-android/issues/49#issuecomment-338713435 - unfortunately, the issues do not always match the issue title :)

**Comment by csb0730** [comment]
Hi @r10s,
these two issues are not the same from technical point of view. Different title is ok.

This issue here describes the situation where a message arrives (new at local device) at _contact requests_ but no indication mark is set because another application (see my description above) has set the IMAP read flag at mail server more quick than DC get's the message locally.

IMHO #49 is related to message dispatching, this one here is message signaling. That's a different case.

**Comment by r10s** [comment]
Okay, however, the original issue is solved with the upcoming release; contact request of known senders or other Delta Chat clients pop up atop of the message list. If you click on them, you'll be ask **do you want to start a chat with ...** and you can answer **Yes**, **Not now** or **Never**.

There may be some fine-tuning needed (please open a new issue as needed), but I think this is the way to go.

![contact-request](https://user-images.githubusercontent.com/9800740/32990413-3f1666e6-cd29-11e7-8c25-42274f699a05.png)

**Comment by csb0730** [comment]
Hi @r10s,
the following behaviour I see now with v0.9.9 (f-droid version):
- New message has been arrived at "Contact Requests"
- No notification for that raised (even what You described at Nov 19 not seen)
- Option to show "Contact Requests" in chatlist by extended configuration option doesn't exist any more (planned feature now?!).

I think this issue should be reopened again or a new issue should be raised?!
Please give me a hint how to handle at best.

**Comment by testbird** [comment]
My try to categorize the typical use cases, variants and related settings:


In this issue, it seems to be "chat and email" in the multi-user variant.

----
Use-Cases
**chat and email** (handle both email and chat exclusively, full MUA mode)
* needs to promt/notify on all new (non-spam) messages (not just known contacts)

**chat only** (share account with external MUA(s)  without disturbing each other)
* disable prompting for emails (long messages)
* needs a SIEVE filter set up on IMAP server moving "Chat:" messages into the chat folder instead of the inbox and watching the chat folder, to avoid disturbing MUAs. This would make sense as a general default for all use-cases.

**intermittent chat/email**  (prompt and (sender specific) rules determine thread handling)
* also prompt on long (email) messages, if no specific rule is defined yet
----
Variants
**multi-user** (email account shared)
* notify user on new messages (already read by others) instead of only "new and unread"

**Comment by testbird** [comment]
edited above

**Comment by csb0730** [comment]
These use cases are interesting and important in some way.
But regarding the core issue I think it's necessary to open a new issue as @r10s suggested at Nov 19.
See new issue ...

**Comment by testbird** [comment]
Thanks, I wrongly posted the above here. See #229 now.
